src/pcre2_compile.c

```diff
@@ -7189,8 +7209,10 @@ for (;; pptr++)
       PUT2INC(code, 0, index);
       PUT2INC(code, 0, count);
       if ((options & PCRE2_CASELESS) != 0)
-        *code++ = ((xoptions & PCRE2_EXTRA_CASELESS_RESTRICT) != 0)?
-            REFI_FLAG_CASELESS_RESTRICT : 0;
+        *code++ = (((xoptions & PCRE2_EXTRA_CASELESS_RESTRICT) != 0)?
+            REFI_FLAG_CASELESS_RESTRICT : 0) |
+                  (((xoptions & PCRE2_EXTRA_TURKISH_CASING) != 0)?
+            REFI_FLAG_TURKISH_CASING : 0);
```

Comment:
I would increase the indentation of `REFI_FLAG_` lines. If the two options follow each other, then a `(xoptions >> log2(opt)) & 0x3` could work as well.